### PR TITLE
Rework fcitx dispatching

### PR DESCRIPTION
**Pull request by berberman**
Summary of changes:
* Call fcitx eventloop on JVM side
* Support dispatching java runnable to fcitx thread
* Eliminate usages of `p_dispatcher` in native code
* Make all fcitx operations suspend current coroutine and dispatch in fcitx thread
* Make `Fcitx` not inherit `CoroutineScope`, since this could be ambiguous when creating coroutines
* Migrate code use suspend fcitx functions 
* Introduce Timber logging
* Implement fake Android lifecycle for Fcitx instance
* Fix concurrency issues

### CODE REVIEW

File: app/src/main/java/me/rocka/fcitx5test/service/FcitxDaemon.kt [hunk 1]
```diff
@@ -1,40 +1,34 @@
...
-                launch {
+class FcitxDaemon : LifecycleService() {
+    private val fcitx by lazy {
+        Fcitx(this).also {
+            lifecycleScope.launch {
+                it.lifecycle.whenStarted {
+                    Timber.d("onReady")
                     while (leftoverOnReadyListeners.isNotEmpty())
-                        leftoverOnReadyListeners.remove()()
+                        leftoverOnReadyListeners.remove().invoke()
```
Comment (berberman) [inline app/src/main/java/me/rocka/fcitx5test/service/FcitxDaemon.kt:22]: What's the purpose of this change?

Comment (rocka) [reply app/src/main/java/me/rocka/fcitx5test/service/FcitxDaemon.kt:22]: `()()` looks confusing to me 😢
